### ISSUE TITLE
"Invalid ELF note descsz" for iris_dri.so-20.1.1-1.fc32.x86_64.debug

### ISSUE THREAD

**Issue by mstange** [gimli-rs/object#238]
When trying to obtain the build ID for [iris_dri.so-20.1.1-1.fc32.x86_64.debug](https://storage.googleapis.com/profiler-get-symbols-fixtures/iris_dri.so-20.1.1-1.fc32.x86_64.debug.gz), `elf_file.build_id()` returns `Err(Error("Invalid ELF note descsz"))`.

`llvm-readelf --notes` on the same file works:

```
~/.mozbuild/clang/bin/llvm-readelf --notes /Users/mstange/Downloads/iris_dri.so-20.1.1-1.fc32.x86_64.debug

Displaying notes found at file offset 0x000002e0 with length 0x00000020:
  Owner                 Data size	Description
  GNU                   0x00000010	NT_GNU_PROPERTY_TYPE_0 (property note)
    Properties:    x86 feature: IBT, SHSTK


Displaying notes found at file offset 0x00000300 with length 0x00000024:
  Owner                 Data size	Description
  GNU                   0x00000014	NT_GNU_BUILD_ID (unique build ID bitstring)
    Build ID: ff62ea37c479f6641e538dc1bb820a4a4baa0ab5

[...]
```

**Comment by bjorn3** [comment]
The error comes from https://github.com/gimli-rs/object/blob/2ad508810e5470f5bcb7ede8d866c8b1eda4c4d3/src/read/elf/note.rs#L70

**Comment by philipc** [comment]
I can't access that file link (says "Not Found"). I recently fixed a note error in #234. Is this before or after that change?

**Comment by mstange** [comment]
Oops, fixed the link. It's probably before that change; will test.

**Comment by mstange** [comment]
This is indeed fixed in object 0.20. Nothing to see here. Thanks! (I'm using `addr2line::object`, so the fix will trickle down to me once there's a new addr2line release that uses object 0.20.)
